test/cache/map_cache_test.dart

```diff
@@ -81,5 +81,21 @@ main() {
 
       expect(count, equals(1));
     });
+
+    test("should not cache a failed request", () async {
+      int count = 0;
+
+      Future<String> loader(String key) {
```

Comment:
nit you can write this a little more simply using async.

Future<String> loader(String key) async {
  count += 1;
  throw "Request failed";
}